### PR TITLE
706 refactor updateconfigproperty to remove mutation

### PR DESCRIPTION
**Pull request by heatherlogan-scottlogic**
## Description

Smol refactor based on Peter's nice suggestions on ticket 🤓
Didn't think it was necessary to check oldValue=newValue in controller as the slider's don't send requests on unchanged values  

## Screenshots

Where applicable, show screenshots of the visual changes.

## Notes

- Bullet point list with
- any notable code changes
- or explanations regarding this PR

## Concerns

- Bullet point list
- with any questions
- or concerns about this PR

## Checklist

Have you done the following?

- [x] Linked the relevant Issue
- [ ] Added tests
- [x] Ensured the workflow steps are passing

### CODE REVIEW

File: backend/src/controller/modelController.ts [hunk 1]
```diff
@@ -37,32 +36,23 @@ function handleConfigureModel(req: OpenAiConfigureModelRequest, res: Response) {
 	const configId = req.body.configId as MODEL_CONFIG | undefined;
 	const value = req.body.value;
 
-	let updated = null;
-
-	if (configId && value && value >= 0) {
+	if (configId && value !== undefined) {
```
Comment (gsproston-scottlogic) [inline backend/src/controller/modelController.ts:39]: Good catch, just checked and this does indeed fix a bug in dev where we couldn't set 0 values for these.

File: backend/src/controller/modelController.ts [hunk 2]
```diff
@@ -37,32 +36,23 @@ function handleConfigureModel(req: OpenAiConfigureModelRequest, res: Response) {
...
-				break;
-			case MODEL_CONFIG.PRESENCE_PENALTY:
-				updated = updateConfigProperty(lastConfig, configId, value, 2);
-				break;
-			default:
-				res.status(400).send();
-		}
-		if (updated) {
-			req.session.chatModel.configuration = updated;
+		const maxValue = configId === MODEL_CONFIG.TOP_P ? 1 : 2;
```
Comment (gsproston-scottlogic) [inline backend/src/controller/modelController.ts:41]: I'd personally be tempted to move this above the `if` statement on line 39, and then add a range check to that `if` statement. 
Then, you don't even need the `updateConfigProperty` method anymore, just do 

```typescript
function handleConfigureModel(req: OpenAiConfigureModelRequest, res: Response) {
	const configId = req.body.configId as MODEL_CONFIG | undefined;
	const value = req.body.value;
	const maxValue = configId === MODEL_CONFIG.TOP_P ? 1 : 2;

	if (configId && value !== undefined && value >= 0 && value <= maxValue) {
		req.session.chatModel.configuration[configId] = value;
		res.status(200).send();
	} else {
		res.status(400).send();
	}
}
```

**Review by gsproston-scottlogic — CHANGES_REQUESTED** [review]
Nicely trimmed down. I actually just thought of a way we could simplify it even further.

**Review by chriswilty — COMMENTED** [review]
What George said! Otherwise good to go.

File: backend/src/controller/modelController.ts (hunk 2 above)
Comment (heatherlogan-scottlogic) [reply backend/src/controller/modelController.ts:41]: ah yeah much nicer!

**Review by gsproston-scottlogic — APPROVED** [review]
Brill!

**Review by chriswilty — APPROVED** [review]
Beautiful!

**Pull request merged**